<?xml version="1.0"?>

<launch>

  <arg name="world_name" default="terrain"/>

  <arg name="laser_enabled" default="true"/>
  <arg name="kinect_enabled" default="false"/>

  <arg name="init_x" default="0"/>
  <arg name="init_y" default="-50"/>

  <!-- launch world file -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find rcomv_uav)/worlds/$(arg world_name).world"/> <!-- world_name is wrt GAZEBO_RESOURCE_PATH environment variable -->
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <!-- launch robot  -->
  <include file="$(find husky_gazebo)/launch/spawn_husky.launch">
    <arg name="laser_enabled" value="$(arg laser_enabled)"/>
    <arg name="kinect_enabled" value="$(arg kinect_enabled)"/>
    <arg name="x" default="$(arg init_x)"/>
    <arg name="y" default="$(arg init_y)"/>
    <arg name="yaw" default="0"/>
  </include>

    <!--- Run Move Base -->  
  <include file="$(find rcomv_ugv)/launch/move_base.launch">
    <arg name="no_static_map" value="true"/>
  </include>

  <!-- path publisher -->
  <arg name="path" default="[0, -50, 5]"/>
  <node pkg="rcomv_ugv" type="circular_path_publisher" name="circular_motion_planner"  output="screen" launch-prefix="xterm -e">
	<rosparam param="path" subst_value="True">$(arg path)</rosparam>
	<param name="number_segements" value="6"/>
	<param name="x" value="$(arg init_x)"/>
    	<param name="y" value="$(arg init_y)"/>
  </node>

</launch>
